### DISCUSSION TITLE
Kafka Listener Loadbalancer type only for bootstrap endpoint

### DISCUSSION THREAD

**Discussion by ntrvic** [strimzi/strimzi-kafka-operator#5913] [Q&A]
I am trying to use the loadBalancer listener type for my Kafka cluster. However since its a large cluster a loadbalancer gets spun up for every single broker in the cluster. I have tired several configurations and tried to find docs but to no avail. Is it possible or supported to just spin up a bootstrap external listener without spinning up a load balancer for every broker(pod) in the cluster?

**Comment by scholzj** [comment]
So, what would you use to expose the brokers? Node Ports? Or how would you want to use it?

**Reply by ntrvic** [reply]
I still have an internal listener for all my brokers. As for external I figured that the "bootstrap" service thats gets spun up targets my pool of brokers using the strimzi-cluster selector label. I don't think the bootstrap service has anything to do with the LBs/nodeports allocated for the individual brokers does it?

**Reply by scholzj** [reply]
The bootstrap service selects all the brokers. But with Kafka, that is not enough. The bootstrap service is used by the clients to get the initial metadata. But afterwards they need direct connection to each individual broker. So having only the bootstrap Load balancer will not get you anywhere. You need to expose the brokers. 

With the `type: loadbalancer` listener, it uses loadbalancer to expose each broker. What you can for example do is to use `type: nodeport` to use node ports to expose the brokers and then manually add a load balancer for bootstrapping. But you need to have more than just the load balancer.

**Reply by ntrvic** [reply]
yes good point, thanks for clarifying. Node port is a good option then with a little added manual step. Thanks for the feedback.
